### PR TITLE
Add cmake python utils

### PR DESCRIPTION
**Pull request by andistorm**
## Changes
* Add functions to create targets for
  * pip_install_local
  * pip_install_dist
  * build_wheel
  * install_wheel
* Add functions/macros to use python virtualvenv in cmake

## TODO after merge
- [ ] Release/Tag new version

### CODE REVIEW

File: python-package-helpers.cmake [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    cmake_parse_arguments(
+        "EV_CREATE_PIP_INSTALL_DIST_TARGET"
+        ""
+        "PACKAGE_NAME"
+        "DEPENDS"
+        ${ARGN}
+    )
+
+    add_custom_target(${EV_CREATE_PIP_INSTALL_DIST_TARGET_PACKAGE_NAME}_pip_install_dist
+        # Remove buil dir from pip
```
Comment (hikinggrass) [inline python-package-helpers.cmake:11]: typo: build dir

File: python-package-helpers.cmake [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+                "No install prefix for wheel specified, please set ${PROJECT_NAME}_WHEEL_INSTALL_PREFIX, use the INSTALL_PREFIX argument or use macro ev_setup_cmake_variables_python_wheel() to set a default value."
+            )
+        else()
+            set(EV_CREATE_PYTHON_WHEEL_TARGETS_INSTALL_PREFIX ${${PROJECT_NAME}_WHEEL_INSTALL_PREFIX})
+        endif()
+    endif()
+
+    set(WHEEL_OUTDIR ${CMAKE_CURRENT_BINARY_DIR}/dist)
+    add_custom_target(${EV_CREATE_PYTHON_WHEEL_TARGETS_PACKAGE_NAME}_build_wheel
+        # Remove buil dir from pip
```
Comment (hikinggrass) [inline python-package-helpers.cmake:88]: same as above

File: python-package-helpers.cmake (hunk 1 above)
Comment (andistorm) [inline python-package-helpers.cmake:11]:
```suggestion
        # Remove build dir from pip
```

File: python-package-helpers.cmake [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    cmake_parse_arguments(
+        "EV_CREATE_PIP_INSTALL_LOCAL_TARGET"
+        ""
+        "PACKAGE_NAME"
+        "DEPENDS"
+        ${ARGN}
+    )
+
+    add_custom_target(${EV_CREATE_PIP_INSTALL_LOCAL_TARGET_PACKAGE_NAME}_pip_install_local
+        # Remove buil dir from pip
```
Comment (andistorm) [inline python-package-helpers.cmake:35]:
```suggestion
        # Remove build dir from pip
```

File: python-package-helpers.cmake (hunk 2 above)
Comment (andistorm) [inline python-package-helpers.cmake:88]:
```suggestion
        # Remove build dir from pip
```

**Review by hikinggrass — APPROVED** [review]

**Pull request merged**